packages/shr-models/lib/models.js

```diff
@@ -754,6 +754,40 @@ class Identifier {
   get namespace() { return this._namespace; }
   get name() { return this._name; }
   get fqn() { return (this.isPrimitive || this.isSpecialKeyWord) ? this._name : `${this._namespace}.${this._name}`; }
+  get title() {
```

Comment:
I don't have time to review this in too much detail right now, but I just want to say that I appreciate that you wrote this out in a logical way distinct from the regular expression I previously used elsewhere. This, although larger, will likely be more easily maintainable than that regex.

Assuming that someone can test this and it works well, good job! And good job in progress, otherwise. :)